lib/WWW/Mechanize.pm

```diff
@@ -1432,11 +1439,16 @@ sub all_forms_with_fields {
 }
 
 
-=head2 $mech->form_with_fields( @fields )
+=head2 $mech->form_with_fields( @fields, [$nth] )
```

Comment:
I'm not very familiar with HTML, but can we assume that field names do not consist of digits only?
That is, can we assume that the following form is invalid?
```html
<form action="http://localhost/" method="post" enctype="multipart/form-data" name="4th_form_2">
    <input type="text" name="4a"  />
    <input type="text" name="2"  /> <!-- DIGIT ONLY FIELD NAME -->
    <input type="Submit" name="submit" value="Submit" label="Sumbit" />
</form>
```

Reply:
@jculverhouse did you have thoughts on the comments above?

Reply:
@skaji I just checked and https://validator.w3.org doesn't have an issue with the input names in your example form.

Reply:
If `name="\d+"` is valid,
https://gist.github.com/skaji/e6445cb1c33dc89d5be7565d51eb2121 shows that form_with_fields() of this PR is ambiguous.

If we really want to specify "nth" form, I would suggest using a hash `{ n => "nth" }` like select() method.

Reply:
Good catch @skaji and also a good suggestion for the API.